jaeger-zipkin/src/main/java/com/uber/jaeger/senders/zipkin/ThriftSpanConverter.java

```diff
@@ -78,13 +78,39 @@
     List<LogData> logs = span.getLogs();
     if (logs != null) {
       for (LogData logData : logs) {
-        annotations.add(new Annotation(logData.getTime(), logData.getMessage()));
+        String logMessage = logData.getMessage();
+        Map<String, ?> logFields = logData.getFields();
+        if (logMessage != null) {
+          annotations.add(new Annotation(logData.getTime(), logMessage));
+        } else if (logFields != null) {
+          annotations.add(new Annotation(logData.getTime(), logFieldsAsMessage(logFields)));
+        }
       }
     }
 
     return annotations;
   }
 
+  private static String logFieldsAsMessage(Map<String, ?> logFields) {
```

Comment:
Even better. Will switch to a JSON encoding for this. Looks like Gson is on the classpath already.